src/main/java/com/teragrep/aer_02/json/JsonResourceId.java

```
@@ -0,0 +1,106 @@
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.aer_02.json;
+
+import jakarta.json.*;
+import jakarta.json.stream.JsonParsingException;
+
+import java.io.StringReader;
+import java.util.Objects;
+
+public final class JsonResourceId {
+
+    private final String message;
+
+    public JsonResourceId(final String message) {
+        this.message = message;
+    }
+
+    public String resourceId() {
+        final String resourceIdKey = "resourceId";
+        final JsonStructure mainStructure;
+        try (final JsonReader reader = Json.createReader(new StringReader(message))) {
+            mainStructure = reader.read();
+        }
+        catch (JsonParsingException e) {
+            // Could not parse JSON, return empty id
+            return "";
+        }
+
+        if (mainStructure == null || !mainStructure.getValueType().equals(JsonValue.ValueType.OBJECT)) {
+            // No main structure or it wasn't a JSONObject, return empty id
+            return "";
+        }
+
+        final JsonObject mainObject = mainStructure.asJsonObject();
+
+        if (!mainObject.containsKey(resourceIdKey)) {
+            // No "resourceId" key present in JSONObject, return empty id
+            return "";
+        }
+
+        if (!mainObject.get(resourceIdKey).getValueType().equals(JsonValue.ValueType.STRING)) {
+            // resourceId was not a string, return empty id
+            return "";
+        }
```

It feels like these checks for invalid JSON data should throw an exception instead of returning an empty string, or is it required to have them return an empty string in case JSON parsing errors?

---

The idea here was that the received event would always be sent, even if it is not in the expected format.